libs/platform/src/lib/components/form/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,104 @@
+import { NgControl, NgForm } from '@angular/forms';
+import { Component, ChangeDetectorRef, Self, Optional, Input, ViewChild, AfterViewInit } from '@angular/core';
+import { CheckboxComponent as CoreCheckboxComponent, stateType } from '@fundamental-ngx/core';
+import { BaseInput } from '../base.input';
+import { Status } from './../form-control';
+
+let uniqueId = 0;
+
+@Component({
+    selector: 'fdp-checkbox',
+    templateUrl: './checkbox.component.html',
+})
+export class CheckboxComponent extends BaseInput implements AfterViewInit {
+    /** reference of child checkbox implementation */
+    @ViewChild(CoreCheckboxComponent)
+    corecheckbox: CoreCheckboxComponent;
+
+    /** set to true if binary checkbox */
+    @Input()
+    binary: boolean = false;
+
+    /** Sets label for checkbox. */
+    @Input()
+    label: string;
+
+    /** indeterminate state */
+    @Input()
+    tristate: boolean = false;
+
+    /** indeterminate state can be selected  */
+    @Input()
+    tristateSelectable: boolean = true;
+
+    /** value for checkbox, when it is not binary */
+    @Input()
+    get value(): any {
+        return super.getValue();
+    }
+    set value(value: any) {
+        if (value) {
+            this._value = value;
+        }
+    }
+
+    /** set status value , will be coming from checkbox group*/
+    set stateStatus(newStatus: Status) {
+        this._status = newStatus;
+        this._cd.markForCheck();
+    }
+
+    /** returns form status */
+    get stateStatus(): Status {
+        return this._status;
+    }
+
+    /** @hidden captures controller value */
+    private modelValue: any;
+
+    /** tracking checkbox current value */
+    checkboxValue: any;
+
+    constructor(
+        protected _changeDetector: ChangeDetectorRef,
+        @Optional() @Self() public ngControl: NgControl,
+        @Optional() @Self() public ngForm: NgForm
+    ) {
+        super(_changeDetector, ngControl, ngForm);
+
+        if (this.ngControl) {
+            this.ngControl.valueAccessor = this;
+        }
+
+        // @hidden have to set default initial values as base class has check and throws error
+        this.id = `fdp-checkbox-id-${uniqueId++}`;
```

Comment:
let's remove this one, this is why we have it in parent. and the goal is to let app developer set it. and if not then have some default value, but we should not do this. this is really a fallback in case you dont sent it. I am seeing this more and more and we should remove this code from our compoennts

Reply:
This has been removed. default id is getting set from baseInput.